src/main/scala/wdl4s/MemorySize.scala

```diff
@@ -0,0 +1,39 @@
+package wdl4s
+
+import wdl4s.parser.MemoryUnit
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+object MemorySize {
+  val memoryPattern = """(\d+(?:\.\d+)?)\s*(\w+)""".r
+  def parse(unparsed: String): Try[MemorySize] = {
```

Comment:
I actually think "chucking X on unsuspecting consumers" is exactly why Try is better than throwing an Exception.

I know I personally am much happier handling a Try than belatedly realising that some code I'm using throws exceptions and having to re-wrap it in a Try() myself